openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/RequestUtil.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.openmrs.projectbuendia.webservices.rest;
+
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.projectbuendia.Utils;
+
+import javax.annotation.Nullable;
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * Utilities for working with requests and request parameters.
+ */
+public class RequestUtil {
+    /**
+     * Obtains the "Sync From" date from a request. This is stored in the "since" HTTP parameter
+     * in ISO 8601 format.
+     * @param context the relevant request.
+     * @return the instant that data should be provided from (inclusive), or {@code null} if the
+     * "since" parameter was not present in the request.
+     */
+    @Nullable
+    public static Date getSyncFromDate(RequestContext context) throws ParseException {
+        String param = context.getParameter("since");
+        if (param == null) {
+            return null;
+        }
+        return Utils.fromIso8601(param);
```

Comment:
Perhaps it's just another difference in taste?  I hate/worry about time zone issues so much (and have been bitten by costly time zone bugs so many times) that using a format in which it is _impossible_ to express anything other than UTC is worth a heckuva lot to me.  To me avoiding this class of bugs greatly outweighs the (subjectively) small cost of converting millis to y/m/d/h/m/s when I'm debugging.

It's also nice that every database and programming language knows how to store and handle integers, but y/m/d/h/m/s structs are handled in lots of different ways and many systems have no primitive.  For example, 2016-02-29T00:00:00Z is parseable but 2015-02-29T00:00:00Z is not, and whether 2015-12-19T00:00:60Z is parseable is implementation-dependent.  If one can look at data on the wire and be unable to tell whether it's well-formed, that feels pretty tragic.